src/iface/ethernet.rs

```diff
@@ -1174,16 +1445,54 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
             f(ip_repr, payload)
         })
     }
+
+    #[cfg(all(feature = "proto-ipv4", feature = "proto-igmp"))]
+    fn igmp_report_packet<'any>(&self, version: IgmpVersion, group_addr: Ipv4Address) -> Option<Packet<'any>> {
+        self.ipv4_address().map(|iface_addr| {
+            let igmp_repr = IgmpRepr::MembershipReport {
+                group_addr,
+                version,
+            };
+            let pkt = Packet::Igmp((Ipv4Repr {
+                src_addr:    iface_addr.clone(),
+                // Send to the group being reported
+                dst_addr:    group_addr,
+                protocol:    IpProtocol::Igmp,
+                payload_len: igmp_repr.buffer_len(),
+                hop_limit:   1,
```

Comment:
From #60:
> the TTL value for group query has to be set to 1, and to 255 for membership report (according to Wireshark and RFC 3717)

Reply:
I don't like the semantics of TTL=255, that is: reporting group membership to every group listener across routers. I'm not seeing it in the wild either.

As this is IGMPv2 being implemented here, not [IP over Optical Networks: A Framework](https://tools.ietf.org/html/rfc3717)(?), from [RFC 2236](https://tools.ietf.org/html/rfc2236):
> All IGMP messages described in this document are sent with IP TTL 1, and contain the IP Router Alert option [RFC 2113] in their IP header.
> […]
> When a group's timer expires, the host multicasts a Version 2 Membership Report to the group, with IP TTL of 1.

Ok, I didn't know about the Router Alert option… but I guess the comment regarding TTL might have been a mistake.